Add CMake build system

rtburns-jpl opened the pull request:
This commit adds experimental CMake build support for ISCE2.

The only changes involved should be adding CMakeLists.txt files.
All current source files, headers, and other functionality, including
the existing SCons build system, should be unaffected.

Some functionality is still a work-in-progress.
These should all be marked as TODO in the CMakeLists.txt files.
`grep TODO **.txt`

Please read CMakeLists.txt and .cmake/*.cmake for more info.

rtburns-jpl committed:
Add CMake build system

This commit adds experimental CMake build support for ISCE2.

The only changes involved should be adding CMakeLists.txt files.
All current source files, headers, and other functionality, including
the existing SCons build system, should be unaffected.

Some functionality is still a work-in-progress.
These should all be marked as TODO in the CMakeLists.txt files.
`grep TODO **.txt`

Please read CMakeLists.txt and .cmake/*.cmake for more info.

rtburns-jpl committed:
Fix X11 lib detection

rtburns-jpl committed:
Add main X11 to mdx link libraries

rtburns-jpl committed:
Merge branch 'master' into cmake

rtburns-jpl committed:
Fix rpath by installing shared libs to system path

rtburns-jpl committed:
Add isceobj.Sensor.ScanSAR

rtburns-jpl committed:
Add demStitch extension module

rtburns-jpl committed:
Add isce2 applications to $PATH

rtburns-jpl committed:
Install mdx executable to bin

rtburns-jpl committed:
Fix typo

rtburns-jpl committed:
Update dependencies

rtburns-jpl committed:
Merge commit '64768d4' into cmake

rtburns-jpl committed:
Merge commit 'f95a912' into cmake

rtburns-jpl committed:
Merge commit 'ed30cba' into cmake

rtburns-jpl committed:
Merge commit 'c92759a' into cmake

rtburns-jpl commented:
This should now be synced with ALOS 2 functionality from @CunrenLiang's PRs #107 and #111, as well as scansar removal from #109.

rtburns-jpl committed:
Add alos2proc_f python package initializer

piyushrpt commented:
Whenever, this is done - think we should update .circleci/config.yml and just repeat build, install and test with the cmake version as well?

rtburns-jpl commented:
Yeah that sounds good. I noticed some leftover test files while I was setting this up so I added testing commands for them - most are failing but some might be suitable for automation.

rtburns-jpl commented on contrib/alos2proc_f/src/cbind.f90:
```diff
@@ -28,7 +28,8 @@ subroutine c_rect(infile,outfile,ndac,nddn,nrac,nrdn,a,b,c,d,e,f,filetype,intsty
 
-    subroutine c_rect_with_looks(infile,outfile,ndac,nddn,nrac,nrdn,a,b,c,d,e,f,lac,ldn,lac0,ldn0,filetype,intstyle) bind(c, name="c_rect_with_looks")
+    subroutine c_rect_with_looks(infile,outfile,ndac,nddn,nrac,nrdn,a,b,c,d,e,f,lac,ldn,lac0,ldn0,filetype,intstyle) &
+            bind(c, name="c_rect_with_looks")
```
Had to wrap this line to avoid fortran -Werror=truncation. For some reason -ffixed-line-length-none did not prevent this.

rtburns-jpl committed:
Use exact library output name

Alos2filter expects a .so library, but macOS defaults to .dylib,
so we must override the library prefix/name/suffix.

rtburns-jpl committed:
Merge branch 'master' into cmake

rtburns-jpl committed:
Add rtcApp

rtburns-jpl committed:
Add preliminary circleci config

rtburns-jpl committed:
Merge branch 'master' into cmake

rtburns-jpl committed:
Remove unused cython includedir logic

rtburns-jpl committed:
Enable -fPIC for StdOEL common lib

rtburns-jpl committed:
Add fftwf to alos2filter libs

rtburns-jpl committed:
Use python module dir

rtburns-jpl committed:
Fix ctest workdir

rtburns-jpl committed:
Disable failing tests

rtburns-jpl committed:
Use conda ctest

rtburns-jpl committed:
Install to conda prefix

rtburns-jpl committed:
Disable primary scons job temporarily

rtburns-jpl committed:
Build verbosely

rtburns-jpl committed:
Test using isce2 application path

rtburns-jpl committed:
Fix typo

rtburns-jpl committed:
Use relative python module path

rtburns-jpl committed:
Revert "Disable primary scons job temporarily"

This reverts commit 786f205fe8387c991afb9fb7762ebfa2804b51ac.

rtburns-jpl commented:
Circleci is very cool 😎 

@piyushrpt I think this is ready if you'd like to check it out. The cmake job uses the same base image + deps as the scons job so we might be able to make them share initial steps with some more advanced circleci-fu

rtburns-jpl committed:
Merge branch 'master' into cmake

rtburns-jpl committed:
Remove unused lib

piyushrpt commented:
There have been many changes since this PR started. Do we want to rebase this? 
I also noticed in circleci/config that we are not testing for alos2App and new stuff that have been added. Will add separate PR for that. We can possibly reuse the same for cmake

rtburns-jpl commented:
I'm fine to rebase/flatten but not sure if that would help. I've tried to keep up the changes as they are merged in but ultimately I think that will only be reliable with more thorough testing.

piyushrpt commented:
@rtburns-jpl merge this in when you feel like this is ready. I will start playing with it this weekend.
